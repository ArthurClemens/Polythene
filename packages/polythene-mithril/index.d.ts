export { Button } from "polythene-mithril-button";
export { ButtonGroup } from "polythene-mithril-button-group";
export { Card } from "polythene-mithril-card";
export { Checkbox } from "polythene-mithril-checkbox";
export { Dialog } from "polythene-mithril-dialog";
export { DialogPane } from "polythene-mithril-dialog-pane";
export { Drawer } from "polythene-mithril-drawer";
export { FAB } from "polythene-mithril-fab";
export { Icon } from "polythene-mithril-icon";
export { IconButton } from "polythene-mithril-icon-button";
export { IOSSpinner } from "polythene-mithril-ios-spinner";
export { List } from "polythene-mithril-list";
export { ListTile } from "polythene-mithril-list-tile";
export { MaterialDesignProgressSpinner } from "polythene-mithril-material-design-progress-spinner";
export { MaterialDesignSpinner } from "polythene-mithril-material-design-spinner";
export { Menu } from "polythene-mithril-menu";
export { RadioButton } from "polythene-mithril-radio-button";
export { RadioGroup } from "polythene-mithril-radio-group";
export { Shadow } from "polythene-mithril-shadow";
export { SVG } from "polythene-mithril-svg";
export { Switch } from "polythene-mithril-switch";
