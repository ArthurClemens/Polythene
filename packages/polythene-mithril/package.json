{
  "name": "polythene-mithril",
  "version": "1.4.2",
  "description": "",
  "main": "dist/polythene-mithril",
  "module": "dist/polythene-mithril.mjs",
  "scripts": {
    "clean": "rm -rf dist/*",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/bundling/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/bundling/rollup.es.js",
    "build-standalone": "ENTRY=scripts/standalone.js DEST=./dist/polythene-mithril-standalone ../../node_modules/rollup/bin/rollup -c ../../scripts/bundling/rollup.browser.js",
    "build": "npm run clean && npm run rollup && npm run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-css": "^1.4.2",
    "polythene-mithril-base": "^1.4.2",
    "polythene-mithril-button": "^1.4.2",
    "polythene-mithril-button-group": "^1.4.2",
    "polythene-mithril-card": "^1.4.2",
    "polythene-mithril-checkbox": "^1.4.2",
    "polythene-mithril-dialog": "^1.4.2",
    "polythene-mithril-dialog-pane": "^1.4.2",
    "polythene-mithril-drawer": "^1.4.2",
    "polythene-mithril-fab": "^1.4.2",
    "polythene-mithril-icon": "^1.4.2",
    "polythene-mithril-icon-button": "^1.4.2",
    "polythene-mithril-ios-spinner": "^1.4.2",
    "polythene-mithril-list": "^1.4.2",
    "polythene-mithril-list-tile": "^1.4.2",
    "polythene-mithril-material-design-progress-spinner": "^1.4.2",
    "polythene-mithril-material-design-spinner": "^1.4.2",
    "polythene-mithril-menu": "^1.4.2",
    "polythene-mithril-notification": "^1.4.2",
    "polythene-mithril-radio-button": "^1.4.2",
    "polythene-mithril-radio-group": "^1.4.2",
    "polythene-mithril-raised-button": "^1.4.2",
    "polythene-mithril-ripple": "^1.4.2",
    "polythene-mithril-search": "^1.4.2",
    "polythene-mithril-shadow": "^1.4.2",
    "polythene-mithril-slider": "^1.4.2",
    "polythene-mithril-snackbar": "^1.4.2",
    "polythene-mithril-svg": "^1.4.2",
    "polythene-mithril-switch": "^1.4.2",
    "polythene-mithril-tabs": "^1.4.2",
    "polythene-mithril-textfield": "^1.4.2",
    "polythene-mithril-toolbar": "^1.4.2"
  },
  "devDependencies": {
    "polythene-core": "^1.4.2",
    "polythene-core-css": "^1.4.2"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "homepage": "https://github.com/ArthurClemens/polythene",
  "license": "MIT"
}
