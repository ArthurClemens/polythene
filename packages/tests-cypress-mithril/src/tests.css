.page__test__component .align-items-center {
  display: flex;
  align-items: center;
}