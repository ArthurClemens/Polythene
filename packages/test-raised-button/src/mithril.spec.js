import { runSnapshots } from "../../../scripts/tests/mithril-snapshots";
import { renderer, keys, Button } from "polythene-mithril";
import specTests from "./spec-tests.js";
import mithrilTests from "./tests-mithril.js";

runSnapshots({
  tests: specTests({ Button, renderer, keys }).concat(mithrilTests),
  renderer
});
