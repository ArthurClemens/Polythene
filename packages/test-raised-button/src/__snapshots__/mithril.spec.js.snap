// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button row 1`] = `
"<div class=\\"pe-button-row\\">
  <a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=
  \\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      One
    </div>
  </div></a><a class=\\"pe-button pe-text-button pe-raised-button\\"
  tabindex=\\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      Two
    </div>
  </div></a><a class=\\"pe-button pe-text-button pe-raised-button\\"
  tabindex=\\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      Three
    </div>
  </div></a>
</div>
"
`;

exports[`Dark tone class + light tone 1`] = `
"<div style=\\"background: rgb(255, 255, 255);\\" class=\\"pe-button-row\\">
  <a class=
  \\"pe-button pe-text-button pe-raised-button pe-light-tone\\"
  tabindex=\\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      Normal
    </div>
  </div></a><a class=
  \\"pe-button pe-text-button pe-raised-button pe-button--disabled pe-light-tone\\"
  tabindex=\\"-1\\" disabled>
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-0\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-0\\"></div>
    </div>
    <div class=\\"pe-button__label\\">
      Disabled
    </div>
  </div></a><a class=
  \\"pe-button pe-text-button pe-raised-button pe-light-tone tests-raised-button-themed-button\\"
  tabindex=\\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      Theme
    </div>
  </div></a>
</div>
"
`;

exports[`Dark tone class + light tone class 1`] = `
"<div style=\\"background: rgb(255, 255, 255);\\" class=
\\"pe-button-row pe-light-tone\\">
  <a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=
  \\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      Normal
    </div>
  </div></a><a class=
  \\"pe-button pe-text-button pe-raised-button pe-button--disabled\\"
  tabindex=\\"-1\\" disabled>
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-0\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-0\\"></div>
    </div>
    <div class=\\"pe-button__label\\">
      Disabled
    </div>
  </div></a><a class=
  \\"pe-button pe-text-button pe-raised-button tests-raised-button-themed-button\\"
  tabindex=\\"0\\">
  <div class=\\"pe-button__content\\">
    <div class=\\"pe-shadow pe-shadow--animated\\">
      <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
      <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
    </div>
    <div class=\\"pe-ripple\\"></div>
    <div class=\\"pe-button__wash\\"></div>
    <div class=\\"pe-button__focus\\"></div>
    <div class=\\"pe-button__label\\">
      Theme
    </div>
  </div></a>
</div>
"
`;

exports[`No options 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
</div></a>
"
`;

exports[`Option: animateOnTap (false) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Don't animate shadow
  </div>
</div></a>
"
`;

exports[`Option: children 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div><span>Child</span>
</div></a>
"
`;

exports[`Option: class 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button class-x\\"
tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
</div></a>
"
`;

exports[`Option: className 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button className-x\\"
tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
</div></a>
"
`;

exports[`Option: content 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div><span>Content</span>
</div></a>
"
`;

exports[`Option: disabled (false) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Not disabled
  </div>
</div></a>
"
`;

exports[`Option: disabled (true) 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button pe-button--disabled\\"
tabindex=\\"-1\\" disabled>
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-0\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-0\\"></div>
  </div>
  <div class=\\"pe-button__label\\">
    Disabled
  </div>
</div></a>
"
`;

exports[`Option: disabled -- dark tone class 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button pe-button--disabled\\"
tabindex=\\"-1\\" disabled>
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-0\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-0\\"></div>
  </div>
  <div class=\\"pe-button__label\\">
    Label
  </div>
</div></a>
"
`;

exports[`Option: element 1`] = `
"<button class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=
\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
</div></button>
"
`;

exports[`Option: id 1`] = `
"<a id=\\"id-x\\" class=\\"pe-button pe-text-button pe-raised-button\\"
tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
</div></a>
"
`;

exports[`Option: inactivate (2s) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Inactivated for 2s
  </div>
</div></a>
"
`;

exports[`Option: inactive (false) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Not inactive
  </div>
</div></a>
"
`;

exports[`Option: inactive (true) 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button pe-button--inactive\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Inactive
  </div>
</div></a>
"
`;

exports[`Option: ink (false) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    No ink
  </div>
</div></a>
"
`;

exports[`Option: label -- dark tone class (should be app's primary color) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Label
  </div>
</div></a>
"
`;

exports[`Option: label 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Label
  </div>
</div></a>
"
`;

exports[`Option: raised (with option z: 2) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-2\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-2\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Raised to 2
  </div>
</div></a>
"
`;

exports[`Option: raised (with option z: 5) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-5\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-5\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Raised to 5
  </div>
</div></a>
"
`;

exports[`Option: selected 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button pe-button--selected\\"
tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Selected
  </div>
</div></a>
"
`;

exports[`Option: style (colors) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div style=
\\"background-color: rgb(239, 108, 0); color: rgb(255, 255, 255);\\"
class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Styled
  </div>
</div></a>
"
`;

exports[`Option: tabindex 1`] = `
"<a tabindex=\\"3\\" class=\\"pe-button pe-text-button pe-raised-button\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
</div></a>
"
`;

exports[`Option: url (with oncreate) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\"
href=\\"#!/shadow\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Go to /#/shadow
  </div>
</div></a>
"
`;

exports[`Option: wash (false) 1`] = `
"<a class=\\"pe-button pe-text-button pe-raised-button\\" tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    No wash
  </div>
</div></a>
"
`;

exports[`Themed button (hover color) 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button tests-raised-button-hover-button\\"
tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Hover button
  </div>
</div></a>
"
`;

exports[`Themed button (should be red) 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button tests-raised-button-themed-button\\"
tabindex=\\"0\\">
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-1\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-1\\"></div>
  </div>
  <div class=\\"pe-ripple\\"></div>
  <div class=\\"pe-button__wash\\"></div>
  <div class=\\"pe-button__focus\\"></div>
  <div class=\\"pe-button__label\\">
    Themed button
  </div>
</div></a>
"
`;

exports[`Themed button (with option disabled) 1`] = `
"<a class=
\\"pe-button pe-text-button pe-raised-button pe-button--disabled tests-raised-button-themed-button\\"
tabindex=\\"-1\\" disabled>
<div class=\\"pe-button__content\\">
  <div class=\\"pe-shadow pe-shadow--animated\\">
    <div class=\\"pe-shadow__bottom pe-shadow--z-0\\"></div>
    <div class=\\"pe-shadow__top pe-shadow--z-0\\"></div>
  </div>
  <div class=\\"pe-button__label\\">
    Disabled themed button
  </div>
</div></a>
"
`;
